src/webview/js/handleInboundEvents.js

```diff
@@ -0,0 +1,44 @@
+/* @flow strict-local */
+import invariant from 'invariant';
+import { ensureUnreachable } from '../../generics';
+
+import type { EditSequence } from '../generateInboundEventEditSequence';
+
+// TODO: Put more of the WebView's event-handling code in here.
+
+/**
+ * PRIVATE - don't use until we have more test coverage (coming soon)
+ */
+export function applyEditSequence(editSequence: EditSequence) {
+  const msglistElementsDiv = document.querySelector('div#msglist-elements');
+  invariant(msglistElementsDiv, 'applyEditSequence: expected msglistElementsDiv');
+
+  editSequence.forEach(edit => {
+    switch (edit.type) {
+      case 'insert': {
+        const fragment = document.createDocumentFragment();
+        const newChild = document.createElement('div');
+        fragment.appendChild(newChild);
+        newChild.outerHTML = edit.html;
+        msglistElementsDiv.insertBefore(fragment, msglistElementsDiv.children[edit.index]);
+        break;
+      }
+      case 'replace': {
+        const fragment = document.createDocumentFragment();
+        const newChild = document.createElement('div');
+        fragment.appendChild(newChild);
+        newChild.outerHTML = edit.html;
+        msglistElementsDiv.replaceChild(fragment, msglistElementsDiv.children[edit.index]);
```

Comment:
Here the fragment again seems unnecessary. In this case it also makes the code look suspicious -- we're replacing the old child with, not the new child, but something wrapping the new child. I believe the wrapping successfully dissolves on insert, but that's less obvious than it could be by not having it in the first place.